staging/src/k8s.io/apiserver/pkg/cel/openapi/schemas.go

```diff
@@ -0,0 +1,264 @@
+/*
```

Comment:
Eventually the one that uses `structural.Structural` would go away, in one of the two ways:
- convert `structural.Structural` to `*spec.Schema`, which we already have https://github.com/kubernetes/kubernetes/blob/v1.26.0/staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/schema/kubeopenapi.go#L24
- extract common interfaces of two types.

Reply:
In the mean time. Because both the CRD validating feature and ValidatingAdmissionPolicy feature have the same group of people as frequent contributors, reviewers, owners, and approvers. Given that OpenAPI schema implementation do not expect much change, we can keep these in sync manually while reviewing.

Reply:
> Eventually the one that uses structural.Structural would go away, in one of the two ways:
>
> * convert structural.Structural to *spec.Schema, which we already have https://github.com/kubernetes/kubernetes/blob/v1.26.0/staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/schema/kubeopenapi.go#L24

I would recommend doing that conversion and unifying on constructing CEL types off of *spec.Schema first, not cloning 3+ large and intricate files and trying to reconverge later.

Reply:
that would also make all the existing CRD CEL validation tests exercise this path, which would build confidence in its correctness

Reply:
I've removed old implementations for Structural. I left original tests intact to avoid breaking anything. However, the CRD validating code couples too strongly with `Structural`, so I kept anything outside the library unchanged with a conversion.